src/engine/SCons/EnvironmentValues.py

```diff
@@ -3,7 +3,7 @@
 _is_valid_var = re.compile(r'[_a-zA-Z]\w*$')
 
 _rm = re.compile(r'\$[()]')
-_remove = re.compile(r'\$\([^\$]*(\$[^\)][^\$]*)*\$\)')
+_remove = re.compile(r'\$\([^$]*(\$[^)][^$]*)*\$\)')
```

Comment:
pretty sure this isn't a good change.   I think remove is meant to remove $X ${X}  and yield X.

Reply:
regex changes are directly from pycharm, which presumably has a good evaluator in it. I'll look at these, however.  Nothing would match your pattern anyway, since it's specc'd to start with literal `$(`.